vpn-proxy/app/models.py

```diff
@@ -135,3 +153,82 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         self.stop()
         super(Tunnel, self).delete(*args, **kwargs)
+
+
+class Ports(models.Model):
+	src_addr = models.GenericIPAddressField(protocol='IPv4')
+	dst_addr = models.GenericIPAddressField(protocol='IPv4', validators=[check_destination_ip])
+	dst_port = models.IntegerField()
+	tunel_id = models.IntegerField()
+	loc_port = models.IntegerField(unique=True)
+	created_at = models.DateTimeField(auto_now_add=True)
+
+	@property
+	def tunnel(self):
+		return '%s%s' % (IFACE_PREFIX, self.tunel_id)
+
+	@property
+	def port(self):
+		return self.loc_port
+
+	@property
+	def rtable(self):
+		return 'rt_%s' % self.tunnel
+
+	@property
+	def destination(self):
+		return '%s:%s' % (self.dst_addr, self.dst_port)
+
+	def __str__(self):
+		return '%s at local port %s via %s -> %s:%s' % (self.src_addr, self.port, self.tunnel, self.dst_addr, self.dst_port)
+
+	def to_dict(self):
+		return {
+			'id': self.id,
+			'src_addr': self.src_addr,
+			'dst_addr': self.dst_addr,
+			'dst_port': self.dst_port,
+			'dst_pair': self.destination,
+			'tunnel_id': self.tunel_id,
+			'tunnel_name': self.tunnel,
+			'loc_port': self.loc_port,
+			'r_table': self.rtable
+		}
+
+	def forward(self, mode, **new_record):
```

Comment:
Also this shouldn't accept `**new_record` arguments, these are instance attributes.